src/commands/build.cr

```diff
@@ -44,9 +44,12 @@ module Shards
         Log.debug { "#{Shards.crystal_bin} #{args.join(' ')}" }
 
         error = IO::Memory.new
-
         status = Process.run(Shards.crystal_bin, args: args, output: Process::Redirect::Inherit, error: error)
-        raise Error.new("Error target #{target.name} failed to compile:\n#{error}") unless status.success?
+        unless status.success?
```

Comment:
Minor nitbit: I prefer to never use `unless` with `else` and instead swap both branches. The success case is also much more common, another reason to have it first.

However, I'd even suggest to leave this line as before as an early exit via raise and just continue in the main body of the method with the success flow.